File: July 2025/23.cpp
class Solution {
  public:
    int subarraySum(vector<int>& arr) {
        long long totalSum = 0;
        int n = arr.size();
        for (int i = 0; i < n; ++i) {
            totalSum += static_cast<long long>(arr[i]) * (i + 1) * (n - i);
        }
        return static_cast<int>(totalSum);
    }
};

File: July 2025/23.py
class Solution:
    def subarraySum(self, arr):
        total_sum = 0
        n = len(arr)
        for i in range(n):
            # Each arr[i] contributes to (i+1)*(n-i) subarrays
            total_sum += arr[i] * (i + 1) * (n - i)
        return total_sum
